virtcontainers/device/drivers/generic.go

```diff
@@ -27,38 +30,49 @@ func NewGenericDevice(devInfo *config.DeviceInfo) *GenericDevice {
 
 // Attach is standard interface of api.Device
 func (device *GenericDevice) Attach(devReceiver api.DeviceReceiver) error {
-	if device.DeviceInfo.Hotplugged {
-		return nil
-	}
-
+	device.AttachCount++
 	return nil
 }
 
 // Detach is standard interface of api.Device
 func (device *GenericDevice) Detach(devReceiver api.DeviceReceiver) error {
-	if !device.DeviceInfo.Hotplugged {
-		return nil
-	}
-
+	device.AttachCount--
 	return nil
 }
 
-// IsAttached checks if the device is attached
-func (device *GenericDevice) IsAttached() bool {
-	return device.DeviceInfo.Hotplugged
+// DeviceType is standard interface of api.Device, it returns device type
+func (device *GenericDevice) DeviceType() config.DeviceType {
+	return config.DeviceGeneric
+}
+
+// GetDeviceInfo returns device information used for creating
+func (device *GenericDevice) GetDeviceInfo() interface{} {
+	return device.DeviceInfo
+}
+
+// GetAttachCount returns how many times the device has been attached
+func (device *GenericDevice) GetAttachCount() int {
+	return device.AttachCount
 }
 
 // DeviceID returns device ID
 func (device *GenericDevice) DeviceID() string {
 	return device.ID
 }
 
-// DeviceType is standard interface of api.Device, it returns device type
-func (device *GenericDevice) DeviceType() config.DeviceType {
-	return config.DeviceGeneric
+// HostPath return host path of device
+func (device *GenericDevice) GetHostPath() string {
+	return device.DeviceInfo.HostPath
 }
 
-// GetDeviceInfo returns device information used for creating
-func (device *GenericDevice) GetDeviceInfo() interface{} {
-	return device.DeviceInfo
+// Reference adds one reference to device
```

Comment:
I'd remove these comments as you've already handled them in the `Device` interface comments.

Reply:
Let's me check whether lint is OK without this comments

Reply:
Just tried. Golint won't allow this, so have to keep a duplicate comment here.